### ISSUE TITLE
Copy_to returning raw dictionary

### ISSUE THREAD

**Issue by juliusdanek** [burnash/gspread#1017]
**Is your feature request related to a problem? Please describe.**
It would be great if we were able to initialize a worksheet from a json. Or alternatively have the [worksheet.copy_to](https://docs.gspread.org/en/latest/api/models/worksheet.html#gspread.worksheet.Worksheet.copy_to) method return the worksheet instead of the raw dictionary returned by google

Below the example of a dict returned by the copy_to method. I would love to be able to immediately edit this worksheet.

json
```
{'sheetId': 1234556, 'title': 'titel', 'index': 6, 'sheetType': 'GRID', 'gridProperties': {'rowCount': 1003, 'columnCount': 25, 'frozenRowCount': 1, 'hideGridlines': True, 'columnGroupControlAfter': True}, 'tabColor': {'red': 0.9882353, 'green': 0.49019608, 'blue': 0.4117647}, 'tabColorStyle': {'rgbColor': {'red': 0.9882353, 'green': 0.49019608, 'blue': 0.4117647}}}
```

**Comment by lavigne958** [comment]
I will look at it. Usually the returned value is the actual data or some value you expect (title, row count etc...) Or the returned value from the API in any other cases.

You JSON sample does not include any data is that expected ?

**Comment by juliusdanek** [comment]
No, it includes all the data expected. It would just be more handy to immediately have a Worksheet object to work with instead of the dictionary.

**Comment by lavigne958** [comment]
Ho I see, what you expect is gpread filters the return dict and removes the metadata etc from google and only return the actual worksheet.

This can be achieved but it must be controlled by a flag, if it currently returns the whole response, then someone may be relies on that return value. we must change this behavior (using a new flag as method parameter)

**Comment by lavigne958** [comment]
I worked on that for some time, I faced an issue, the current protected range have some proprietary IDs that I can't workout and it can't be copied from a sheet to another.

It is interesting to copy them by running the `add_protected_range` method again but it might be costly :thinking:

For sure it will deserve a dedicated issue.

**Comment by juliusdanek** [comment]
Yeah, it's costly unfortunately.

**Comment by lavigne958** [comment]
Hi, I looked at your issue and I am sorry but we can't make it without extra costs.

The `dict` object returned by the API contains only the new sheet properties. it does not contain any informations about the spreadsheet where this sheet has been copied to.

I face 2 issues:
1. with only the sheet properties I could return a new `Worksheet` instance **but** we don't have the parent `Spreadsheet` object that will do all the HTTP requests
2. the method `copy_to` takes the `Spreadsheet` ID so we could do something but in the end it will only be like any user could do: 1) open the destination spreadsheet 2) open the new worksheet.

About the protected ranges this is a good point, but we will tackle it in a different issue, see #1046 

Closing this issue.
